wasm/src/lib.rs

```diff
@@ -37,6 +38,48 @@ pub fn git_hash() -> String {
     git_hash.into()
 }
 
+fn get_qir_internal(code: &str) -> std::result::Result<String, (Error, Vec<Frame>)> {
+    let core = compile::core();
+    let mut store = PackageStore::new(core);
+    let std = compile::std(&store, TargetProfile::Base);
+    let std = store.insert(std);
+    let sources = SourceMap::new([("test".into(), code.into())], None);
+
+    let unit = qsc::compile::compile(
+        &store,
+        &[std],
+        sources,
+        PackageType::Exe,
+        TargetProfile::Base,
+    );
+
+    // Ensure it compiles before trying to add it to the store.
+    if !unit.1.is_empty() {
+        // This should never happen, as the program should be checked for errors before trying to
+        // generate code for it. But just in case, simply report the entire source as 'fail to generate output'
+        return Err((
+            Error::OutputFail(Span {
+                lo: 0,
+                hi: code.len() as u32,
+            }),
+            vec![],
+        ));
+    }
+
+    let package = store.insert(unit.0);
+
```

Comment:
It's clearer to unwrap the tuple here.
```suggestion
    let (unit, errors) = qsc::compile::compile(
        &store,
        &[std],
        sources,
        PackageType::Exe,
        TargetProfile::Base,
    );

    // Ensure it compiles before trying to add it to the store.
    if !errors.is_empty() {
        // This should never happen, as the program should be checked for errors before trying to
        // generate code for it. But just in case, simply report the entire source as 'fail to generate output'
        return Err((
            Error::OutputFail(Span {
                lo: 0,
                hi: code.len() as u32,
            }),
            vec![],
        ));
    }

    let package = store.insert(unit);

```

Reply:
Yeah, I meant to make that change for readability. (Though I do note we often use the tuple.index pattern a bit in some places). Will do. Thanks.